src/typecheck/mod.rs

```diff
@@ -616,6 +616,17 @@ fn type_check_<S, E>(
             envs.insert(x.clone(), ty_let);
             type_check_(state, envs, lin, linearizer, strict, rt, ty)
         }
+        Term::LetPattern(x, pat, re, rt) => {
```

Comment:
@ysndr do we need to do something special here, with respect to the LSP?

Reply:
Maybe add a new new declaration for the variables brought to scope here otherwise they are more or less ignored.

Reply:
Ok, will see